pypose/module/dynamics.py

```diff
@@ -0,0 +1,368 @@
+from numpy import vectorize
+import torch as torch
+import torch.nn as nn
+from torch.autograd.functional import jacobian
+
+class _System(nn.Module):
+    r'''
+    A sub-class of :obj:`torch.nn.Module` to build general dynamics.
+    
+    Args:
+        time (:obj:`boolean`): Whether the system is time-varying; defaults to False, meaning time-invariant
+
+    Linearization
+    ----------
+    The nonlinear state-space equation is given as:
+
+    .. math::
+        \begin{align}
+        \overrightarrow{x}_{k+1} &= \mathbf{f}(\overrightarrow{x}_k,\overrightarrow{u}_k) \\
+        \overrightarrow{y}_{k} &= \mathbf{g}(\overrightarrow{x}_k,\overrightarrow{u}_k)
+        \end{align}
+
+    This class provides a means to linearize the system at any point along a trajectory.
+    \bf{Note}: The linearization can be provided for any arbitrary point, not just the equilibrium point(s).
+

```

Comment:
See https://github.com/pypose/pypose/blob/dmpc_linearDynamics/pypose/module/dynamics.py for the syntax for note.
```
Note:
    blah
```